src/allotropy/parsers/biorad_bioplex/biorad_bioplex_structure.py

```diff
@@ -0,0 +1,268 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Optional
+from xml.etree import ElementTree
+
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.parsers.utils.values import (
+    get_val_from_xml,
+)
+
+WELLS_TAG = "Wells"
+DOC_LOCATION_TAG = "NativeDocumentLocation"
+DESCRIPTION_TAG = "Description"
+PLATE_DIMENSIONS_TAG = "PlateDimensions"
+TOTAL_WELLS_ATTRIB = "TotalWells"
+VERSION_ATTRIB = "BioPlexManagerVersion"
+SAMPLES = "Samples"
+MEMBER_WELLS = "MemberWells"
+ROW_NUMBER = "RowNo"
+COLUMN_NUMBER = "ColNo"
+WELL_NUMBER = "WellNumber"
+WELL_NO = "WellNo"
+REGION_NUMBER = "RegionNumber"
+REGIONS_OF_INTEREST = "RegionsOfInterest"
+CODE = "Code"
+DILUTION = "Dilution"
+BEAD_COUNT = "BeadCount"
+ROW_NAMES = "ABCDEFGH"
+
+SAMPLE_ROLE_TYPE_MAPPING = {
+    "Blank": "blank role",
+    "Control": "control sample role",
+    "Standard": "standard sample role",
+    "Unknown": "unknown sample role",
+}
+
+
+@dataclass(frozen=True)
+class AnalyteSample:
+    analyte_name: str
+    analyte_region: int
+    analyte_error_code: int
+
+    @staticmethod
+    def create(analyte_xml: ElementTree.Element) -> AnalyteSample:
+        return AnalyteSample(
+            analyte_name=str(analyte_xml[0].text),
+            analyte_region=int(analyte_xml.attrib[REGION_NUMBER]),
+            analyte_error_code=int(analyte_xml[1].attrib[CODE]),
+        )
+
+
+@dataclass(frozen=True)
+class WellAnalyteMapping:
+    well_name: str
+    analytes: list[AnalyteSample] = field(default_factory=list)
+
+
+@dataclass
+class SampleDocument:
+    sample_type: str
+    sample_identifier: str
+    description: str
+    well_name: str
+    sample_dilution: float
+    well_analyte_mapping: WellAnalyteMapping
+
+
+@dataclass
+class SampleDocumentAggregate:
+    # This data class pulled from the <Samples> part of the xml.
+    samples: list[SampleDocument] = field(default_factory=list)
+    # Default to empty dictionary.
+    analyte_region_dict: dict[int, str] = field(default_factory=dict)
+
+    @staticmethod
+    def create(samples_xml: ElementTree.Element) -> SampleDocumentAggregate:
+        sample_documents = SampleDocumentAggregate()
+        for sample_types in samples_xml:
+            for child_sample_type in sample_types:
+                # TODO: add catch of non mapping sample roles
```

Comment:
Would you be able to just do this now? You can write a getter func that throws an AllotropeConversionError and call it good :D